keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,1044 @@
+while there are jobs with the custom `managed-by` label, can be indicative of
+the built-in job controller stepping onto another controller, causing conflicts.
+This can be further investigate per-job by checking the `.metadata.managedFields.manager`
+being flipped between two owners.
+
+The feature is opt-in so in case of such problems the custom `managed-by` label
+should not be used.
+
+A substantial drop in the `job_sync_duration_seconds`, while the number of
+jobs with the custom `managed-by` label is low, could be indicative of the
+Job controller skipping reconciliation of jobs it should reconcile. This could
+be further investigated per-job by looking at the timestamp of changes in
+`.metadata.managedFields.time`, and owners in `.metadata.managedFields.manager`.
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+It will be tested manually prior to beta launch.
+
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+Check if a Job has the `managed-by` label.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ ] Events
+  - Event Reason:
+- [X] API .metadata
+  - Condition name:
+  - Other field:
+    - `.metadata.labels['batch.kubernetes.io/managed-by']` for Jobs
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+This feature does not propose SLOs.
```

Comment:
I'd expect the SLO to be consistent with the current SLO for the job controller, not sure if we have them defined. But we need to make sure that it should not get worse.

Reply:
We have some SLO: https://github.com/kubernetes/community/blob/master/sig-scalability/slos/slos.md, but non of them look specific to Job controller. I don't expect any of them being impacted, will add the note.